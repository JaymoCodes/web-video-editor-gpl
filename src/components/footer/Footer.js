import React, { Component } from 'react';
import config from '../../utils/config/config.js';

import withWindowDimensions from '../../hoc/withWindowDimensions.js';
import PropTypes from 'prop-types';
import styles from './Footer.module.css';

class Footer extends Component {

  /*
   * Get current layout, depending on window dimensions
   */
  getFooterClasses = () => {
    const footerClasses = [styles.footer];
    this.props.isMobileSized || this.props.isTabletSized ? footerClasses.push(styles.mobileLayout) : footerClasses.push(styles.desktopLayout);
  
    return footerClasses.join(' ');
  }

  render() {
    return (
      <div className={this.getFooterClasses()}>
        <div className={styles.footerContainer}>
          <div className={styles.logo}>Â© 2021 Accusonus.com</div>
          <div className={styles.linksContainer}>
            <a href={config.PATH.terms} className={styles.link}>{config.FOOTER.terms}</a>
            <a href={config.PATH.patents} className={styles.link}>{config.FOOTER.patents}</a>
            <a href={config.PATH.policy} className={styles.link}>{config.FOOTER.policy}</a>
            <a href={config.PATH.source} className={styles.link}>{config.FOOTER.source}</a>
          </div>
        </div>
      </div>
    );
  }
}

Footer.propTypes = {
  mode: PropTypes.string,
  // Provided by withWindowDimensions() HOC
  windowWidth: PropTypes.number,
  windowHeight: PropTypes.number,
  isMobileSized: PropTypes.bool,
  isTabletSized: PropTypes.bool,
  isDesktopSized: PropTypes.bool,
};

export default withWindowDimensions(Footer);
