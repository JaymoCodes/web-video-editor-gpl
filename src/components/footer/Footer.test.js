import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';

describe('Footer', () => {
  it('Renders without crashing', () => {
    let wrapper;
  
    wrapper = renderer.create(
      <Footer/>
    );
  
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('Mobile rendering', () => {
    let wrapper;
  
    wrapper = renderer.create(
      <Footer
        isMobileSized={true}
        isTabletSized={false}
        isDesktopSized={false}
      />
    );

    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('Tablet rendering', () => {
    let wrapper;
  
    wrapper = renderer.create(
      <Footer
        isMobileSized={false}
        isTabletSized={true}
        isDesktopSized={false}
      />
    );
  
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('Desktop rendering', () => {
    let wrapper;
  
    wrapper = renderer.create(
      <Footer
        isMobileSized={false}
        isTabletSized={false}
        isDesktopSized={true}
      />
    );
  
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
