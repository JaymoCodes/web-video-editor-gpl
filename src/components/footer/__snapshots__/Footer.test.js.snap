// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Desktop rendering 1`] = `
<div
  className="footer desktopLayout"
>
  <div
    className="footerContainer"
  >
    <div
      className="logo"
    >
      © 2021 Accusonus.com
    </div>
    <div
      className="linksContainer"
    >
      <a
        className="link"
        href=""
      >
        Terms of use
      </a>
      <a
        className="link"
        href=""
      >
        Patents
      </a>
      <a
        className="link"
        href=""
      >
        Privacy Policy
      </a>
      <a
        className="link"
        href="https://github.com/accusonus/web-video-editor-gpl"
      >
        Get Source Code
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer Mobile rendering 1`] = `
<div
  className="footer mobileLayout"
>
  <div
    className="footerContainer"
  >
    <div
      className="logo"
    >
      © 2021 Accusonus.com
    </div>
    <div
      className="linksContainer"
    >
      <a
        className="link"
        href=""
      >
        Terms of use
      </a>
      <a
        className="link"
        href=""
      >
        Patents
      </a>
      <a
        className="link"
        href=""
      >
        Privacy Policy
      </a>
      <a
        className="link"
        href="https://github.com/accusonus/web-video-editor-gpl"
      >
        Get Source Code
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer Renders without crashing 1`] = `
<div
  className="footer desktopLayout"
>
  <div
    className="footerContainer"
  >
    <div
      className="logo"
    >
      © 2021 Accusonus.com
    </div>
    <div
      className="linksContainer"
    >
      <a
        className="link"
        href=""
      >
        Terms of use
      </a>
      <a
        className="link"
        href=""
      >
        Patents
      </a>
      <a
        className="link"
        href=""
      >
        Privacy Policy
      </a>
      <a
        className="link"
        href="https://github.com/accusonus/web-video-editor-gpl"
      >
        Get Source Code
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer Tablet rendering 1`] = `
<div
  className="footer mobileLayout"
>
  <div
    className="footerContainer"
  >
    <div
      className="logo"
    >
      © 2021 Accusonus.com
    </div>
    <div
      className="linksContainer"
    >
      <a
        className="link"
        href=""
      >
        Terms of use
      </a>
      <a
        className="link"
        href=""
      >
        Patents
      </a>
      <a
        className="link"
        href=""
      >
        Privacy Policy
      </a>
      <a
        className="link"
        href="https://github.com/accusonus/web-video-editor-gpl"
      >
        Get Source Code
      </a>
    </div>
  </div>
</div>
`;
